library/core-utils/src/test/java/im/vector/lib/core/utils/timer/CountUpTimerTest.kt

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright (c) 2023 New Vector Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package im.vector.lib.core.utils.timer
+
+import im.vector.lib.core.utils.test.fakes.FakeClock
+import io.mockk.every
+import io.mockk.mockk
+import io.mockk.verifySequence
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.test.advanceTimeBy
+import kotlinx.coroutines.test.currentTime
+import kotlinx.coroutines.test.runTest
+import org.junit.Test
+
+private const val AN_INTERVAL = 500L
+private const val AN_INITIAL_TIME = 2_333L
+
+@OptIn(ExperimentalCoroutinesApi::class)
+internal class CountUpTimerTest {
+
+    private val fakeClock = FakeClock()
+
+    @Test
+    fun `when pausing and resuming the timer, the timer ticks the right values at the right moments`() = runTest {
+        every { fakeClock.epochMillis() } answers { currentTime }
```

Comment:
Alternatively we could define a TestClock like this:
```kotlin
@OptIn(ExperimentalCoroutinesApi::class)
class TestClock(private val ts: TestScope) : Clock {
    override fun epochMillis(): Long {
        return ts.currentTime
    }
}
```
(do not change, it's just to share my idea :) ).